### PR TITLE
JCLOUDS-891 Add CloudSigma Miami region

### PR DESCRIPTION
**Pull request by stuarthendren**

### CODE REVIEW

File: cloudsigma2-mia/pom.xml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+
+  <build>
+  	<pluginManagement>
+  		<plugins>
+  			<!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence on the Maven build itself.-->
```
Comment (nacx) [inline cloudsigma2-mia/pom.xml]: IIRC, this is something that can be avoided by fixing it in eclipse? The pom issues I have can be fixed by right clicking the error > Quick Fix > and picking the one marked as (Experimental), which works well.

**Comment by nacx** [comment]
Just one minor comment. We don't put IDE specific configuration in the pom files. Can you check if you can fix it as mentioned without changing the pom? Apart from that lgtm! Thanks @stuarthendren!

**Comment by stuarthendren** [comment]
I don't think those failures are anything to do with this change.

**Comment by nacx** [comment]
Let's see if jenkins is happy now: rebuild please

**Comment by nacx** [comment]
Thanks @stuarthendren! can you squash the commits into a single one so I can cleanly merge it? Also, have you run the live tests against this region? Do you have the output?

**Comment by stuarthendren** [comment]
I have run the live tests, and I get some failures. However, these are the same failures I get for other zones, eg zhr. Which may be because of my set up, or genuine errors. 

For example, I had to go to the UI and specify a currency, before testGetPricing passed, but I think the "It should be in state 'stopped', not 'deleting'" is probably a genuine error.

I could open a separate ticket for these, unless you have set-up suggestions to try?

```
[TestNG] Running:
  C:\Users\stuart\AppData\Local\Temp\testng-eclipse--1446238178\testng-customsuite.xml

Jun 26, 2015 2:56:22 PM org.jclouds.compute.internal.BaseComputeServiceLiveTest testCompareSizes
INFO: smallest {id=cpu=1000.000000,ram=1024, providerId=cpu=1000.000000,ram=1024, processors=[{cores=1.0, speed=1000.0}], ram=1024, volumes=[{type=LOCAL, bootDevice=true, durable=false}], supportsImage=ALWAYS_TRUE}
Jun 26, 2015 2:56:22 PM org.jclouds.compute.internal.BaseComputeServiceLiveTest testCompareSizes
INFO: fastest {id=cpu=40000.000000,ram=1024, providerId=cpu=40000.000000,ram=1024, processors=[{cores=1.0, speed=40000.0}], ram=1024, volumes=[{type=LOCAL, bootDevice=true, durable=false}], supportsImage=ALWAYS_TRUE}
Jun 26, 2015 2:56:22 PM org.jclouds.compute.internal.BaseComputeServiceLiveTest testCompareSizes
INFO: biggest {id=cpu=1000.000000,ram=65536, providerId=cpu=1000.000000,ram=65536, processors=[{cores=1.0, speed=1000.0}], ram=65536, volumes=[{type=LOCAL, bootDevice=true, durable=false}], supportsImage=ALWAYS_TRUE}
Jun 26, 2015 2:57:18 PM org.jclouds.compute.internal.BaseComputeServiceLiveTest createAndRunAServiceInGroup
INFO: << available node(33605de1-b97e-4bb3-a0e9-aeb8a4010061) os(null) in 53s
Jun 26, 2015 2:57:36 PM org.jclouds.compute.internal.BaseComputeServiceLiveTest testGetAssignableLocations
WARNING: location {scope=PROVIDER, id=cloudsigma2-mia, description=https://mia.cloudsigma.com/api/2.0/, iso3166Codes=[US-FL]}
Jun 26, 2015 3:00:53 PM org.jclouds.compute.internal.BaseComputeServiceLiveTest$1 call
INFO: Started node e1958945-2bd8-4b7f-9c44-e97225ec7fa2
Jun 26, 2015 3:00:55 PM org.jclouds.compute.internal.BaseComputeServiceLiveTest$1 call
INFO: Started node b0304f39-3d84-400d-af97-8af19d926204
PASSED: testCalculateSubscriptions
PASSED: testCloneLibraryDrive
PASSED: testCreateDrive
PASSED: testCreateDrives
PASSED: testCreateFirewallPolicies
PASSED: testCreateFirewallPolicy
PASSED: testCreateServer
PASSED: testCreateServers
PASSED: testEditIP
PASSED: testEditProfileInfo
PASSED: testEditVLAN
PASSED: testGetAccountBalance
PASSED: testGetCurrentUsage
PASSED: testGetIPInfo
PASSED: testGetLibraryDrive
PASSED: testGetPricing
PASSED: testGetProfileInfo
PASSED: testGetVLANInfo
PASSED: testListDiscounts
PASSED: testListIPInfo
PASSED: testListIPs
PASSED: testListLibraryDrives
PASSED: testListLicenses
PASSED: testListSubscriptions
PASSED: testListTransactions
PASSED: testListVLANInfo
PASSED: testListVLANs
PASSED: testCloneDrive
PASSED: testCreateTag
PASSED: testCreateTags
PASSED: testDeleteDrives
PASSED: testDeleteServers
PASSED: testEditDrive
PASSED: testEditFirewallPolicy
PASSED: testEditServer
PASSED: testGetDriveInfo
PASSED: testGetFirewallPolicy
PASSED: testGetServerInfo
PASSED: testListDrives
PASSED: testListDrivesInfo
PASSED: testListFirewallPolicies
PASSED: testListFirewallPoliciesInfo
PASSED: testListServers
PASSED: testListServersInfo
PASSED: testDeleteTags
PASSED: deleteFirewallPolicies
PASSED: testDeleteServer
PASSED: testEditTag
PASSED: testGetTagInfo
PASSED: testListTags
PASSED: testDeleteDrive
PASSED: testDeleteTag
PASSED: setServiceDefaults
PASSED: testCompareSizes
PASSED: testCorrectAuthException
PASSED: testCorrectExceptionRunningNodesNotFound
PASSED: testGetAssignableLocations
PASSED: testImageById
PASSED: testImagesCache
PASSED: testListImages
PASSED: testListSizes
PASSED: testOptionToNotBlock
PASSED: testConcurrentUseOfComputeServiceToCreateNodes
PASSED: testTemplateMatch
FAILED: testGetSubscription
org.jclouds.http.HttpResponseException: command: GET https://mia.cloudsigma.com/api/2.0/subscriptions/free_mem/ HTTP/1.1 failed with response: HTTP/1.1 500 INTERNAL SERVER ERROR; content: [[{"error_point": null, "error_type": "backend", "error_message": "unknown error"}]]
	at org.jclouds.cloudsigma2.handlers.CloudSigmaErrorHandler.handleError(CloudSigmaErrorHandler.java:52)
	at org.jclouds.http.handlers.DelegatingErrorHandler.handleError(DelegatingErrorHandler.java:67)
	at org.jclouds.http.internal.BaseHttpCommandExecutorService.shouldContinue(BaseHttpCommandExecutorService.java:136)
	at org.jclouds.http.internal.BaseHttpCommandExecutorService.invoke(BaseHttpCommandExecutorService.java:105)
	at org.jclouds.rest.internal.InvokeHttpMethod.invoke(InvokeHttpMethod.java:90)
	at org.jclouds.rest.internal.InvokeHttpMethod.apply(InvokeHttpMethod.java:73)
	at org.jclouds.rest.internal.InvokeHttpMethod.apply(InvokeHttpMethod.java:44)
	at org.jclouds.rest.internal.DelegatesToInvocationFunction.handle(DelegatesToInvocationFunction.java:156)
	at org.jclouds.rest.internal.DelegatesToInvocationFunction.invoke(DelegatesToInvocationFunction.java:123)
	at com.sun.proxy.$Proxy54.getSubscription(Unknown Source)
	at org.jclouds.cloudsigma2.CloudSigma2ApiLiveTest.testGetSubscription(CloudSigma2ApiLiveTest.java:636)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)

FAILED: testCreateAndRunAService
java.lang.IllegalArgumentException: no credential or ssh agent found for root on node 33605de1-b97e-4bb3-a0e9-aeb8a4010061
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:148)
	at org.jclouds.compute.functions.CreateSshClientOncePortIsListeningOnNode.apply(CreateSshClientOncePortIsListeningOnNode.java:64)
	at org.jclouds.compute.functions.CreateSshClientOncePortIsListeningOnNode.apply(CreateSshClientOncePortIsListeningOnNode.java:40)
	at org.jclouds.compute.callables.SudoAwareInitManager.init(SudoAwareInitManager.java:67)
	at org.jclouds.compute.callables.RunScriptOnNodeAsInitScriptUsingSsh.init(RunScriptOnNodeAsInitScriptUsingSsh.java:64)
	at org.jclouds.compute.callables.RunScriptOnNodeAsInitScriptUsingSshAndBlockUntilComplete.init(RunScriptOnNodeAsInitScriptUsingSshAndBlockUntilComplete.java:77)
	at org.jclouds.compute.callables.RunScriptOnNodeAsInitScriptUsingSshAndBlockUntilComplete.init(RunScriptOnNodeAsInitScriptUsingSshAndBlockUntilComplete.java:38)
	at org.jclouds.compute.internal.BaseComputeService.runScriptOnNode(BaseComputeService.java:640)
	at org.jclouds.compute.internal.BaseComputeServiceLiveTest.createAndRunAServiceInGroup(BaseComputeServiceLiveTest.java:738)
	at org.jclouds.compute.internal.BaseComputeServiceLiveTest.testCreateAndRunAService(BaseComputeServiceLiveTest.java:708)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)

FAILED: testAScriptExecutionAfterBootWithBasicTemplate
org.jclouds.compute.RunScriptOnNodesException: error runScript on filtered nodes options({loginUser=root, loginPasswordPresent=true, loginPrivateKeyPresent=true, runAsRoot=false, wrapInInitScript=false})
Execution failures:

0 error[s]
Node failures:

1) IllegalArgumentException on node 72251668-3a44-4494-ac5e-1f3e2c14ca49:
java.lang.IllegalArgumentException: no credential or ssh agent found for root on node 72251668-3a44-4494-ac5e-1f3e2c14ca49
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:148)
	at org.jclouds.compute.functions.CreateSshClientOncePortIsListeningOnNode.apply(CreateSshClientOncePortIsListeningOnNode.java:64)
	at org.jclouds.compute.functions.CreateSshClientOncePortIsListeningOnNode.apply(CreateSshClientOncePortIsListeningOnNode.java:40)
	at org.jclouds.compute.callables.RunScriptOnNodeUsingSsh.init(RunScriptOnNodeUsingSsh.java:99)
	at org.jclouds.compute.strategy.InitializeRunScriptOnNodeOrPlaceInBadMap.call(InitializeRunScriptOnNodeOrPlaceInBadMap.java:61)
	at org.jclouds.compute.strategy.InitializeRunScriptOnNodeOrPlaceInBadMap.call(InitializeRunScriptOnNodeOrPlaceInBadMap.java:34)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


1 error[s]
	at org.jclouds.compute.internal.BaseComputeService.runScriptOnNodesMatching(BaseComputeService.java:598)
	at org.jclouds.compute.internal.BaseComputeService.runScriptOnNodesMatching(BaseComputeService.java:558)
	at org.jclouds.compute.internal.BaseComputeServiceLiveTest.testAScriptExecutionAfterBootWithBasicTemplate(BaseComputeServiceLiveTest.java:229)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)

FAILED: testCreateTwoNodesWithRunScript
org.jclouds.concurrent.TransformParallelException: error destroyNodesMatching(inGroup(cloudsigma2-)): {{id=e3e67b47-b8a7-4fc4-a85e-4e70e1410034, providerId=e3e67b47-b8a7-4fc4-a85e-4e70e1410034, name=cloudsigma2--aaa, location={scope=PROVIDER, id=cloudsigma2-mia, description=https://mia.cloudsigma.com/api/2.0/, iso3166Codes=[US-FL]}, group=cloudsigma2-, imageId=976368fb-cb6f-43d8-9632-4d19911b9304, status=UNRECOGNIZED, loginPort=22, hardware={id=e3e67b47-b8a7-4fc4-a85e-4e70e1410034, providerId=e3e67b47-b8a7-4fc4-a85e-4e70e1410034, processors=[{cores=1.0, speed=1000.0}], ram=1073741824, volumes=[{id=0945d4d7-15d9-4269-a77f-1be896ec6df7, type=NAS, size=1.07374182E10, bootDevice=true, durable=true}], supportsImage=ALWAYS_TRUE}, userMetadata={image_id=976368fb-cb6f-43d8-9632-4d19911b9304, ssh_public_key=XXXXXXX-REMOVED-XXXXXXXXXXX}}=java.util.concurrent.ExecutionException: org.jclouds.http.HttpResponseException: command: DELETE https://mia.cloudsigma.com/api/2.0/servers/e3e67b47-b8a7-4fc4-a85e-4e70e1410034/ HTTP/1.1 failed with response: HTTP/1.1 403 FORBIDDEN; content: [[{"error_point": null, "error_type": "permission", "error_message": "Error attempting to put 'e3e67b47-b8a7-4fc4-a85e-4e70e1410034' in state 'deleting'. It should be in state 'stopped', not 'deleting'."}]]}
	at org.jclouds.concurrent.FutureIterables.transformParallel(FutureIterables.java:100)
	at org.jclouds.concurrent.FutureIterables.transformParallel(FutureIterables.java:70)
	at org.jclouds.compute.internal.BaseComputeService.destroyNodesMatching(BaseComputeService.java:254)
	at org.jclouds.compute.internal.BaseComputeServiceLiveTest.testCreateTwoNodesWithRunScript(BaseComputeServiceLiveTest.java:336)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)
Caused by: java.util.concurrent.ExecutionException: org.jclouds.http.HttpResponseException: command: DELETE https://mia.cloudsigma.com/api/2.0/servers/e3e67b47-b8a7-4fc4-a85e-4e70e1410034/ HTTP/1.1 failed with response: HTTP/1.1 403 FORBIDDEN; content: [[{"error_point": null, "error_type": "permission", "error_message": "Error attempting to put 'e3e67b47-b8a7-4fc4-a85e-4e70e1410034' in state 'deleting'. It should be in state 'stopped', not 'deleting'."}]]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.jclouds.concurrent.FutureIterables$1.run(FutureIterables.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.jclouds.http.HttpResponseException: command: DELETE https://mia.cloudsigma.com/api/2.0/servers/e3e67b47-b8a7-4fc4-a85e-4e70e1410034/ HTTP/1.1 failed with response: HTTP/1.1 403 FORBIDDEN; content: [[{"error_point": null, "error_type": "permission", "error_message": "Error attempting to put 'e3e67b47-b8a7-4fc4-a85e-4e70e1410034' in state 'deleting'. It should be in state 'stopped', not 'deleting'."}]]
	at org.jclouds.cloudsigma2.handlers.CloudSigmaErrorHandler.handleError(CloudSigmaErrorHandler.java:52)
	at org.jclouds.http.handlers.DelegatingErrorHandler.handleError(DelegatingErrorHandler.java:65)
	at org.jclouds.http.internal.BaseHttpCommandExecutorService.shouldContinue(BaseHttpCommandExecutorService.java:136)
	at org.jclouds.http.internal.BaseHttpCommandExecutorService.invoke(BaseHttpCommandExecutorService.java:105)
	at org.jclouds.rest.internal.InvokeHttpMethod.invoke(InvokeHttpMethod.java:90)
	at org.jclouds.rest.internal.InvokeHttpMethod.apply(InvokeHttpMethod.java:73)
	at org.jclouds.rest.internal.InvokeHttpMethod.apply(InvokeHttpMethod.java:44)
	at org.jclouds.rest.internal.DelegatesToInvocationFunction.handle(DelegatesToInvocationFunction.java:156)
	at org.jclouds.rest.internal.DelegatesToInvocationFunction.invoke(DelegatesToInvocationFunction.java:123)
	at com.sun.proxy.$Proxy54.deleteServer(Unknown Source)
	at org.jclouds.cloudsigma2.compute.strategy.CloudSigma2ComputeServiceAdapter.destroyNode(CloudSigma2ComputeServiceAdapter.java:285)
	at org.jclouds.compute.strategy.impl.AdaptingComputeServiceStrategies.destroyNode(AdaptingComputeServiceStrategies.java:181)
	at org.jclouds.compute.internal.BaseComputeService$2.apply(BaseComputeService.java:291)
	at org.jclouds.compute.internal.BaseComputeService$2.apply(BaseComputeService.java:288)
	at org.jclouds.util.Predicates2$RetryablePredicate.apply(Predicates2.java:117)
	at org.jclouds.compute.internal.BaseComputeService.doDestroyNode(BaseComputeService.java:302)
	at org.jclouds.compute.internal.BaseComputeService$1$1.call(BaseComputeService.java:262)
	at org.jclouds.compute.internal.BaseComputeService$1$1.call(BaseComputeService.java:260)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at java.lang.Thread.getStackTrace(Thread.java:1552)
	at org.jclouds.compute.internal.BaseComputeService$1.apply(BaseComputeService.java:260)
	at org.jclouds.compute.internal.BaseComputeService$1.apply(BaseComputeService.java:255)
	at org.jclouds.concurrent.FutureIterables.transformParallel(FutureIterables.java:82)
	at org.jclouds.concurrent.FutureIterables.transformParallel(FutureIterables.java:70)
	at org.jclouds.compute.internal.BaseComputeService.destroyNodesMatching(BaseComputeService.java:254)
	at org.jclouds.compute.internal.BaseComputeServiceLiveTest.testCreateTwoNodesWithRunScript(BaseComputeServiceLiveTest.java:336)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)

SKIPPED: testCreateTwoNodesWithOneSpecifiedName
SKIPPED: testCreateAnotherNodeWithANewContextToEnsureSharedMemIsntRequired
SKIPPED: testCredentialsCache
SKIPPED: testGet
SKIPPED: testReboot
SKIPPED: testSuspendResume
SKIPPED: testGetNodesWithDetails
SKIPPED: testListNodes
SKIPPED: testListNodesByIds
SKIPPED: testDestroyNodes

===============================================
    Default test
    Tests run: 78, Failures: 4, Skips: 10
===============================================


===============================================
Default suite
Total tests run: 78, Failures: 4, Skips: 10
===============================================

[TestNG] Time taken by org.testng.reporters.XMLReporter@282003e1: 23 ms
[TestNG] Time taken by [FailedReporter passed=0 failed=0 skipped=0]: 17 ms
[TestNG] Time taken by org.testng.reporters.JUnitReportReporter@3e77a1ed: 30 ms
[TestNG] Time taken by org.testng.reporters.EmailableReporter2@62fdb4a6: 21 ms
[TestNG] Time taken by org.testng.reporters.SuiteHTMLReporter@517cd4b: 485 ms
[TestNG] Time taken by org.testng.reporters.jq.Main@5cdd8682: 66 ms
```

**Comment by nacx** [comment]
Pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/10908462) and [1.9.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/a98c77d2). Thanks @stuarthendren and apologies for the delay!

**Pull request closed without merging**